homeworks/dankhazipeter/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,94 @@
+"""Flotta kezelő alkalmazás"""
+
+
+class Car:
+    """
+    Egy autót reprezentáló osztály alapvető funkciókkal,
+    ami kezeli az autó tulajdonságait (márka, modell, év, km, üzemanyag)
+    és műveleteket (vezetés, tankolás).
+    """
+
+    def __init__(self, brand, model, year):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0
+        self.fuel_level = 100
+
+    def drive(self, distance):
+        """
+        Szimulálja az autó haladását a megadott távolságon,
+        ahol minden kilométer 0.1 egység üzemanyagot fogyaszt.
+        """
+        fuel_needed = distance * 0.1
+        if fuel_needed <= self.fuel_level:
+            self.mileage += distance
+            self.fuel_level -= fuel_needed
+        else:
+            max_distance = self.fuel_level / 0.1
+            self.mileage += max_distance
+            self.fuel_level = 0
+            print(f"Nincs elég üzemanyag a(z) {distance} km megtételéhez. "
+                  f"Helyette {max_distance} km-t tett meg.")
+
+    def refuel(self, amount):
+        """Növeli az üzemanyagszintet a megadott mennyiséggel, de legfeljebb 100-ig."""
+        if self.fuel_level + amount > 100:
```

Comment:
Szuper, hogy ellenőrzöd hogy max 100%-ig lehessen tölteni, viszont azt is jó lenne biztosítani, hogy nem használják ezt a metódust arra, hogy leszívják a benzint :)
Jelenleg átengedi a method hogy negatív számot adjak meg a refuel()-nek, ezáltal csökkentsem az üzemanyagszintet.

Reply:
Basszus teljesen jogos és ezzel még 0 alá is lehetett vinni az üzemanyagszintet. Ez súlyos hiba, javítottam.

Reply:
Basszus teljesen jogos és ezzel még 0 alá is lehetett vinni az üzemanyagszintet. Ez súlyos hiba, javítottam.